File: .qoder/skills/get-timestamp/scripts/get-timestamp.py
#!/usr/bin/env python3
"""Get current timestamp in specified format."""
import argparse
from datetime import datetime

FORMATS = {
    "YYYY-MM-DD-HHmm": "%Y-%m-%d-%H%M",
    "YYYY-MM-DD": "%Y-%m-%d",
    "HHmm": "%H%M",
    "ISO": "%Y-%m-%dT%H:%M:%S%z"
}

def get_timestamp(format_name: str = "YYYY-MM-DD-HHmm") -> str:
    """Get current timestamp in specified format."""
    now = datetime.now()
    if format_name == "ISO":
        return now.astimezone().isoformat()
    return now.strftime(FORMATS.get(format_name, "%Y-%m-%d-%H%M"))

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Get current timestamp")
    parser.add_argument(
        "--format", "-f",
        default="YYYY-MM-DD-HHmm",
        help="Output format (YYYY-MM-DD-HHmm, YYYY-MM-DD, HHmm, ISO)"
    )
    args = parser.parse_args()
    print(get_timestamp(args.format))

File: workspace-template/scripts/get-timestamp.js
#!/usr/bin/env node
/**
 * Get current timestamp in specified format
 * Usage: node get-timestamp.js [format]
 * Default format: YYYY-MM-DD-HHmmss
 */

const FORMATS = {
  'YYYY-MM-DD-HHmmss': () => {
    const now = new Date();
    return `${now.getFullYear()}-${String(now.getMonth() + 1).padStart(2, '0')}-${String(now.getDate()).padStart(2, '0')}-${String(now.getHours()).padStart(2, '0')}${String(now.getMinutes()).padStart(2, '0')}${String(now.getSeconds()).padStart(2, '0')}`;
  },
  'YYYY-MM-DD': () => {
    const now = new Date();
    return `${now.getFullYear()}-${String(now.getMonth() + 1).padStart(2, '0')}-${String(now.getDate()).padStart(2, '0')}`;
  },
  'HHmm': () => {
    const now = new Date();
    return `${String(now.getHours()).padStart(2, '0')}${String(now.getMinutes()).padStart(2, '0')}`;
  },
  'ISO': () => new Date().toISOString()
};

function getTimestamp(format = 'YYYY-MM-DD-HHmmss') {
  const formatter = FORMATS[format];
  if (formatter) {
    return formatter();
  }
  // Default fallback
  return FORMATS['YYYY-MM-DD-HHmmss']();
}

// Main execution
const format = process.argv[2] || 'YYYY-MM-DD-HHmmss';
console.log(getTimestamp(format));
